### PR TITLE
TWP-681 Added MediaStream constructor

### PR DESCRIPTION
**Pull request by miniruwan**
Added MediaStream constructor as requested in https://jira.temasys.com.sg/browse/TWP-681

### CODE REVIEW

File: source/adapter.js [hunk 1]
```diff
@@ -561,6 +561,9 @@ createIceServer = null;
 createIceServers = null;
 //------------------------------------------------------------
 
+//Creates MediaStream object.
+MediaStream = null;
```
Comment (letchoo) [inline source/adapter.js]: @miniruwan I think we should not set the `MediaStream` to `null` because this could be defined for other browsers.

**Comment by letchoo** [comment]
Seems ok for me now.

File: source/adapter.js [hunk 2]
```diff
@@ -1036,6 +1040,14 @@ if ( (navigator.mozGetUserMedia ||
         ConstructSessionDescription(info.type, info.sdp);
     };
 
+    MediaStream = function (mediaStreamOrTracks) {
+      AdapterJS.WebRTCPlugin.WaitForPluginReady();
+      if (typeof mediaStreamOrTracks !== 'undefined' && mediaStreamOrTracks !== null) {
```
Comment (johache) [inline source/adapter.js]: What happens in the plugin if you receive undefined or null (might be different on IE and Safari) ?
Could we remove that if statement ?

Comment (miniruwan) [inline source/adapter.js]: @johache yes, we can remove this if statement in javascript.
**Explanation :** In the plugin c++ code, there is an if condition which is true when mediaStreamOrTracks is not available. That if condition is true when mediaStreamOrTracks javascript object is either null or undefined. In that case, a new MediaStream object will be created without any streams in it (as if it were created without any arguments for MediaStream() constructor).

Comment (miniruwan) [inline source/adapter.js]: Done. pushed.

**Comment by johache** [comment]
LGTM

**Pull request merged**